### DISCUSSION TITLE
Why selecting the current version switches to the home page?

### DISCUSSION THREAD

**Discussion by syeda-git** [squidfunk/mkdocs-material#6537] [Q&A]
Hi,
From the CF216 page, when i select the same CF216 version, the current page stays. This seems to be a good behavior.
https://opensource.hcltechsw.com/digital-experience/CF216/build_sites/practitioner_studio/web_content/

However, if i select an h2 in the same page and then select CF216, the page switches to the home page. Any idea why this is happening? 
https://opensource.hcltechsw.com/digital-experience/CF216/build_sites/practitioner_studio/web_content/#hclsoftware-u-learning-materials

Also, why the header title changes from HCL Digital Experience Help Center to Home?
![image](https://github.com/squidfunk/mkdocs-material/assets/44337085/1545f04d-7445-48e4-89c5-543680182c95)

Syed

**Comment by kamilkrzyskow** [comment]
Hello @syeda-git,
as the version control links do indeed point to the home page of that given version, switching to the homepage is the default behaviour.
Changing to the same page on the different alias is an addition of mkdocs-material:
https://github.com/squidfunk/mkdocs-material/issues/4012 
https://github.com/squidfunk/mkdocs-material/blob/05596664a536799c6da8aa708ff8fc7ebeca4337/src/templates/assets/javascripts/integrations/version/index.ts#L93-L146

This could be a bug with the interception or `navigation.instant`, so please create an new issue. https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/
**However,** I see that the linked page uses version `8.5.10` instead of the newest `9.5.2`, so please update the version before submitting the issue. I don't remember every issue posted here, so perhaps that was already fixed along the way.

**Comment by syeda-git** [comment]
@kamilkrzyskow Here's a sample doc built from a recent version of Material for MkDocs. From this page, when you select 2.0 again, it goes to the home page. It doesn't stay on the same page. Is this a bug?
https://syeda-git.github.io/mkdocs-sandbox/2.0/about/

Syed

**Reply by kamilkrzyskow** [reply]
I didn't write the code, nor am I a TypeScript developer, so I'm unsure if it's a bug or intentional 🤔

Given that there is `if (version === current) return EMPTY` and [EMPTY](https://rxjs.dev/api/index/const/EMPTY) points to an ending of sort in the chain  I guess if the current version is the same as the clicked version then it doesn't process it further. So if I'm reading the code correctly this is intended.

If you're unsatisfied with the answer then feel free to follow the bug reporting guide with a minimal example or @ mention squidfunk here to ask him directly. 

Also I guess you could influence the `href` attribute of the links directly with extra JavaScript, though I'm not sure how well it would work given the Observables would still trigger 🤔
https://squidfunk.github.io/mkdocs-material/customization/#additional-javascript
I once wrote a code that recreates the `md-version` toggle and switches between deployments, you might find it useful as reference:
https://github.com/Gothic-Modding-Community/gmc/blob/4bd4fef7ffa6d644a087e544e0c2078dfec0087a/overrides/assets/javascripts/extra.js#L299-L338

**Reply by syeda-git** [reply]
@squidfunk Can you confirm if it is an intended behavior? If yes, is there a way to stop that? If i were to go to the home page, I would click Home.

**Reply by squidfunk** [reply]
I'm not sure from what I've read. However, as you're on 8.5.10, please first update to the latest version. The problem might have been fixed in the meantime. Updating to the latest version is a prerequisite for a complete and helpful bug report anyway.

**Reply by kamilkrzyskow** [reply]
@squidfunk The link provided in the recent message: https://syeda-git.github.io/mkdocs-sandbox/2.0/about/ uses version 9.4.8, maybe not the most recent, but according to git blame the [file](https://github.com/squidfunk/mkdocs-material/blame/master/src/templates/assets/javascripts/integrations/version/index.ts#L93-L146) didn't have any relevant changes since then.

The reported issue is being that when a user is on a version branch `/2.0/` in the example above, and they click the same version from the toggle it won't "reload the page" keeping the address, but instead loads the home page. So the version navigation isn't intercepted to point to same page of the same version it's only intercepted when selecting another version.

Hence the question.
> Can you confirm if it is an intended behavior? If yes, is there a way to stop that? If i were to go to the home page, I would click Home.

**Reply by squidfunk** [reply]
Please create an issue with a minimal reproduction and exact steps to deploy and we look into it. Doesn't sound intended.

**Reply by syeda-git** [reply]
@squidfunk  https://github.com/squidfunk/mkdocs-material/issues/6657
